File: backend/internal/model/cloud_platform.go
// Copyright (c) 2025 kk
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT

package model

import (
	"time"

	"gorm.io/gorm"
)

// CloudPlatform represents a cloud platform (e.g., Aliyun, Tencent Cloud, AWS, Azure)
type CloudPlatform struct {
	ID          uint           `gorm:"primaryKey" json:"id"`
	Name        string         `gorm:"not null;size:100;uniqueIndex" json:"name"`
	Description string         `gorm:"type:text" json:"description"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`

	// Relationships
	Assets []Asset `gorm:"foreignKey:CloudPlatformID" json:"assets,omitempty"`
}

File: frontend/src/api/cloudPlatform.ts
// Copyright (c) 2025 kk
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT

import apiClient from './client'

export interface CloudPlatform {
  id: number
  name: string
  description: string
  created_at: string
  updated_at: string
}

export interface CreateCloudPlatformRequest {
  name: string
  description?: string
}

export interface UpdateCloudPlatformRequest {
  name?: string
  description?: string
}

export const cloudPlatformApi = {
  list: () => apiClient.get<CloudPlatform[]>('/cloud-platforms'),
  get: (id: number) => apiClient.get<CloudPlatform>(`/cloud-platforms/${id}`),
  create: (data: CreateCloudPlatformRequest) => apiClient.post<CloudPlatform>('/cloud-platforms', data),
  update: (id: number, data: UpdateCloudPlatformRequest) => apiClient.put<CloudPlatform>(`/cloud-platforms/${id}`, data),
  delete: (id: number) => apiClient.delete(`/cloud-platforms/${id}`),
}
